beebop/utils.py

```diff
@@ -94,3 +94,29 @@ def replace_filehashes(folder, filename_dict):
                 if f_key in line:
                     line = line.replace(f_key, f_value)
             print(line)
+
+
+def add_query_ref_status(fs, p_hash, filename_dict):
+    ET.register_namespace('', "http://graphml.graphdrawing.org/xmlns")
+    ET.register_namespace('xsi', "http://www.w3.org/2001/XMLSchema-instance")
+    # list of query filenames
+    query_names = list(filename_dict.values())
+    # list of all component graph filenames
+    file_list = []
+    for file in os.listdir(fs.output_network(p_hash)):
+        if fnmatch.fnmatch(file, 'network_component_*.graphml'):
+            file_list.append(file)
+    for filename in file_list:
+        path = os.path.join(fs.output_network(p_hash), filename)
+        xml = ET.parse(path)
+        graph = xml.getroot()
+        nodes = graph.findall(".//{http://graphml.graphdrawing.org/xmlns}node")
+        for node in nodes:
+            name = node.find("./").text
+            child = ET.Element("data")
+            child.set("key", "ref_query")
+            child.text = 'query' if name in query_names else 'ref'
+            node.append(child)
+        ET.indent(xml, space='  ', level=0)
+        with open(path, 'wb') as f:
+            xml.write(f, encoding='utf-8')
```

Comment:
I see in the docs this is required but I always thought xml was defined to be UTF-8, odd that it defaults to us-ascii of all things